### DISCUSSION TITLE
Why on device vs cloud?

### DISCUSSION THREAD

**Discussion by smaldd14** [jomjol/AI-on-the-edge-device#2647] [General]
Hey there! First off, this project  is very cool and I would like to thank @jomjol for authoring!

I just wanted to start a discussion about why you would want to do the ML/AI/computer vision tasks on the device compared to just sending the picture taken to a model you have running in the cloud?

Especially with recent AI enhancements, namely OpenAI's vision capabilities they will soon open API access for (see [here](https://community.openai.com/t/when-will-vision-api-become-available/402441)), as well as the other major players (Google, Amazon, etc.)

It seems like, to me, it would be easier and more reliant to send the image taken to a service you have running and then use the service to run all AI/ML ops?

I'm new to this area, so if there is something I'm missing, please do tell!

**Comment by caco3** [comment]
The original idea started with processing in a docker container. but it got moved to the used all-on-one-device approach.
The performance is enough and it makes it for the user much easier, so he doesn't have to get into cloud, and service costs.
Also he does not have to trust any service for protecting his data well.

Also, running the models in the cloud would not give you better accuracy! The device uses the same model data as we would use in the cloud, so there would be no benefit.
